src/App.tsx

```diff
@@ -1,39 +1,68 @@
-import { useState } from 'react';
+import { useContext, useState } from 'react';
+import LanguageProvider from 'components/LanguageProvider';
 import LocationAutocomplete from 'components/LocationAutocomplete';
+import Select from 'components/Select';
 import WeatherWidget from 'components/WeatherWidget';
-import TemperatureUnitsSelect from 'components/TemperatureUnitsSelect';
+import LANGUAGE_OPTIONS from 'constants/languageOptions';
+import Languages from 'constants/languages';
+import TEMPERATURE_UNITS_OPTIONS from 'constants/temperatureUnitsOptions';
 import TemperatureUnits from 'constants/temperatureUnits';
+import LanguageContext from 'contexts/LanguageContext';
 import Location from 'types/location';
 import * as classes from './styles';
 
+const LANGUAGE_CHOICE_LABEL_ID = 'language-label';
+const LANGUAGE_CHOICE_LABEL = 'Language choice';
 const LOCATION_AUTOCOMPLETE = 'location-select';
-const TEMPERATURE_UNITS_SELECT = 'unit-select';
+const TEMPERATURE_UNITS_LABEL_ID = 'unit-label';
+const TEMPERATURE_UNITS_LABEL = 'Temperature unit';
+
+interface LanguageOption {
+  label: string;
+  value: Languages;
+}
+
+interface TemperatureUnitOption {
+  label: string;
+  value: TemperatureUnits;
+}
 
 function App() {
+  const { selectedLanguage, onSelectLanguage } = useContext(LanguageContext);
```

Comment:
Why do you need it? `App` renders `LanguageProvider`; it means that all components UNDER provider can read from context but not `App` because it's a parent and it's located ABOVE the context)